requre/cassette.py

```diff
@@ -21,14 +21,670 @@
+        self.use_latency = False
+        self.LATENCY_KEY = "latency"
+        self.key: str = "all"
+        self.key_stategy_cls = StorageKeysInspectDefault
+        self.store_arg_debug_metadata = False
+        self.METADATA_ARG_DEBUG_KEY = "log_call_function"
+        self.METADATA_CALLER_LIST = "module_call_list"
+        self.read_key_exact = False
+
+    def get_latency(self, regenerate=True) -> float:
+        """
+        Returns latency from last store event
+
+        :param regenerate: if False, then it will not be changed, just displayed
+        :return: float
+        """
+        current_time = original_time()
+        diff = current_time - self.current_time
+        if regenerate:
+            self.current_time = current_time
+        return diff
+
+    def dump(self, level, key, values, metadata: Dict, cassette: Any):
+        """
+        Store values in proper format to level[key] item
+
+        :param level: parent dict object
+        :param key: item in dict
+        :param values: what to store (return value of function)
+        :param metadata: store metadata to object from upper level
+        :return: None
+        """
+        ds = DataStructure(values)
+        if metadata:
+            ds.metadata = metadata
+        if self.LATENCY_KEY not in ds.metadata:
+            ds.metadata = {self.LATENCY_KEY: self.get_latency()}
+        self.data = ds
+        item = ds.dump()
+        if self.data_type == DataTypes.Dict:
+            level.setdefault(key, {})
+            level[key][self.key] = item
+        elif self.data_type == DataTypes.DictWithList:
+            level.setdefault(key, {})
+            level[key].setdefault(self.key, [])
+            level[key][self.key].append(item)
+        elif self.data_type in [DataTypes.List, DataTypes.DictWithList]:
+            if cassette.storage_file_version <= 1:
+                # Backward compatibility for requre before using DataStructure
+                item = values
+            level.setdefault(key, [])
+            level[key].append(item)
+        elif self.data_type == DataTypes.Dict:
+            level.setdefault(key, {})
+            level[key][self.key] = item
+        else:
+            level[key] = item
+
+    def load(self, level, cassette: Any):
+        """
+        Get data from storage_object and trasform it to DataStructure object.
+
+        :param level: where data are stored
+        :return: output of function
+        """
+        data = None
+        if self.data_type == DataTypes.List:
+            data = DataStructure.create_from_list(level, cassette=cassette)
+        elif self.data_type == DataTypes.Value:
+            data = DataStructure.create_from_value(level)
+        elif self.data_type == DataTypes.Dict:
+            data = DataStructure.create_from_dict(level)
+        elif self.data_type == DataTypes.DictWithList:
+            data = DataStructure.create_from_dict_with_list(level, cassette=cassette)
```

Comment:
Not need to be done here, but:

- What about having it as a strategy?
  - Have multiple classes implementing `create_from`/`dump` behaviour and avoid those long `if-elif`s.